### PR TITLE
Book detail card

### PR DESCRIPTION
**Pull request by briangerez**
## Summary

Layout of book detail card in the book detail screen.

After corrections:
Refactor in navigator route
BookList and Book component located in Library screen.

## Screenshots

[Change!] Show the screenshots of the views you modified.

#### Galaxy S (or similar)
nexus s

#### Pixel 2 (or similar)
nexus 5

#### Pixel 3 or 3XL (or similar)
android max

#### iPhone SE (or 5s)
see

#### iPhone 6/7/8
6ss

#### iPhone 11
ios max

## Trello Card

https://trello.com/c/BnrTqOGg/17-detalle-del-libro

### CODE REVIEW

File: src/app/screens/BookDetail/components/BookCard/index.tsx [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    <View style={styles.bookCardContainer}>
+      <View style={styles.detailContainer}>
+        <Image
+          style={styles.bookImage}
+          resizeMode="contain"
+          source={imageUrl ? { uri: imageUrl } : placeholderBookImg}
+        />
+        <View style={styles.bookDescription}>
+          <Text style={styles.title}>{title}</Text>
+          <Text style={[styles.availability, isAvailable ? styles.available : styles.notAvailable]}>
```
Comment (sfernandez11) [inline src/app/screens/BookDetail/components/BookCard/index.tsx:28]: Same here => `<Text style={[styles.availability, isAvailable && styles.available]}>`

File: src/app/screens/BookDetail/components/BookCard/index.tsx [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+            {isAvailable ? 'Available' : 'Not Available'}
+          </Text>
+          <Text style={styles.author}>{author}</Text>
+          <Text style={styles.author}>{year}</Text>
+          <Text style={styles.author}>{genre}</Text>
+        </View>
+      </View>
+      <Button type="primary" style={styles.buttonMargin}>
```
Comment (sfernandez11) [inline src/app/screens/BookDetail/components/BookCard/index.tsx:36]: Remove `type="primary"`

File: src/app/screens/BookDetail/components/BookCard/index.tsx [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+        ADD TO WISHLIST
+      </Button>
+      <Button type="secondary">RENT</Button>
```
Comment (sfernandez11) [inline src/app/screens/BookDetail/components/BookCard/index.tsx:39]: `<Button secondary>RENT</Button>`

File: src/app/screens/BookDetail/components/BookCard/styles.ts [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+  availability: {
+    fontSize: 17,
+    fontWeight: 'bold'
+  },
+  available: {
+    color: ATLANTIS
+  },
+  notAvailable: {
+    color: VENETIAN_RED
+  },
```
Comment (sfernandez11) [inline src/app/screens/BookDetail/components/BookCard/styles.ts:44]: Use a default style and condition the other:
``` 
  availability: {
    color: VENETIAN_RED,
    fontSize: 17,
    fontWeight: 'bold'
  },
  available: {
    color: ATLANTIS
  },
```

File: src/app/screens/BookDetail/components/BookCard/styles.ts [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+  author: {
+    color: CHARCOAL,
+    fontSize: 15,
+    lineHeight: 12,
+    paddingTop: 5
+  },
+  bookImage: {
+    height: '100%',
+    width: 70
```
Comment (sfernandez11) [inline src/app/screens/BookDetail/components/BookCard/styles.ts:53]: Don't use `'100%'` for images sizes, use Invision image size: `{ width: 69, height: 105 }`

File: src/constants/navigation.ts [hunk 6]
```diff
@@ -1,9 +1,10 @@
 import ImageHeader from '@components/ImageHeader';
-import { WHITE } from '@constants/colors';
+import { WHITE, ALICE_BLUE } from '@constants/colors';
 
 import { HEADER_HEIGHT } from './dimensions';
 
 export const DEFAULT_NAV_OPTIONS = {
+  cardStyle: { backgroundColor: ALICE_BLUE },
```
Comment (sfernandez11) [inline src/constants/navigation.ts:7]: Why this style here???

File: src/app/screens/BookDetail/index.tsx [hunk 7]
```diff
@@ -1,14 +1,21 @@
...
+    <BookCard
+      imageUrl={imageUrl}
+      title={title}
+      author={author}
+      genre={genre}
+      year={year}
+      isAvailable={isAvailable}
+    />
   );
 }
```
Comment (sfernandez11) [inline src/app/screens/BookDetail/index.tsx:20]: ⚠️ This is wrong! You must receive via navigation param the book not hard-code it
```
function BookDetail({ navigation }: Props) {
const book = navigation.getParam('book', {});
...
return <BookCard {...book} />;
}

In the Library/BookList => `const onPress = () => navigation.navigate(BookDetail,  { book });`

File: src/constants/navigation.ts (hunk 6 above)
Comment (briangerez) [reply src/constants/navigation.ts:7]: I used it for the backgroundColor of all screens. https://reactnavigation.org/docs/stack-navigator/#cardstyle 
What do you think?

File: src/app/screens/BookDetail/index.tsx [hunk 8]
```diff
@@ -1,15 +1,14 @@
...
-  return (
-    <TouchableOpacity onPress={handleGoBack}>
-      <Text>Go Back</Text>
-    </TouchableOpacity>
-  );
+function BookDetail() {
+  const route: BookDetailRouteType = useRoute();
+  const { book } = route.params;
+
+  return <BookCard {...book} />;
```
Comment (sfernandez11) [inline src/app/screens/BookDetail/index.tsx:11]: 👏

**Pull request merged**